Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -46,12 +50,47 @@ internal struct BinaryDecoder: Decoder {
 
     private var complete: Bool {return available == 0}
 
-    internal init(forReadingFrom pointer: UnsafePointer<UInt8>, count: Int, extensions: ExtensionMap? = nil) {
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      options: BinaryDecodingOptions,
+      extensions: ExtensionMap? = nil
+    ) {
         // Assuming baseAddress is not nil.
         p = pointer
         available = count
         fieldStartP = p
         self.extensions = extensions
+        self.options = options
+        recursionBudget = options.messageDepthLimit
+    }
+
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      parent: BinaryDecoder
```

Comment:
![screen shot 2017-05-30 at 2 56 24 pm](https://cloud.githubusercontent.com/assets/5279928/26599952/4c20e852-4548-11e7-89c1-292b3113467e.png)

Which seems to say binary didn't change, but JSON go faster with this change?